conan/tools/meson/toolchain.py

```diff
@@ -424,13 +424,13 @@ def _get_extra_flags(self):
         exelinkflags = self._conanfile_conf.get("tools.build:exelinkflags", default=[], check_type=list)
         linker_scripts = self._conanfile_conf.get("tools.build:linker_scripts", default=[], check_type=list)
         linker_script_flags = ['-T"' + linker_script + '"' for linker_script in linker_scripts]
-        defines = self._conanfile.conf.get("tools.build:defines", default=[], check_type=list)
+        defines = self._conanfile_conf.get("tools.build:defines", default=[], check_type=list)
```

Comment:
Check this line @franramirez688, was it a bug or intended?

Reply:
It's a bug. I don't know how I left this. I just reviewed the code again and I saw [the lines above](https://github.com/conan-io/conan/pull/17266/files#diff-0e1e7a3c21f2c036f65b81f8dc72f539f9c76b2b0a031cd5eca0f1de392bd49eR167) this:

```python
self._backend = conanfile.conf.get("tools.meson.mesontoolchain:backend",
                                           default=backend or 'ninja')
```

Please, change it to:
```python
self._backend = self._conanfile_conf.get("tools.meson.mesontoolchain:backend",
                                           default=backend or 'ninja')
```

Reply:
Done